### ISSUE TITLE
Multi facture status Through API or SDK 39% but we have the consent made with user with mfa

### ISSUE THREAD

**Issue by christianarg** [microsoft/Partner-Center-PowerShell#188]
https://partner.microsoft.com/en-us/pcv/security/compliance shows multi facture status Through API or SDK 39% but we have the consent made with user with mfa.
The user has MFA Enforced using the MFA portal. We have implemented this long ago (not recent implementation)
Please help

**Comment by o-l-a-v** [comment]
I'm currently trying out the Get-PartnerUserSignInActivity, maybe you can list out some details with it as well?

```powershell
<#

    .NOTES
        Resources
            Get-PartnerUserSignInActivity
            https://docs.microsoft.com/en-us/powershell/module/partnercenter/get-partnerusersigninactivity

#>


# Connect - Should open a internet browser windows
Connect-PartnerCenter

# Disconnect
Disconnect-PartnerCenter


# Get sign ins 7 days back
$SignIns = Get-PartnerUserSignInActivity -StartDate ([datetime]::Now.AddDays(-7))


# Count by
## All
$SignIns.'Count'
## UPN vs No UPN
$SignIns | Select-Object -Property @{'Name'='UpnPresent';'Expression'={-not[string]::IsNullOrEmpty($_.'UserPrincipalName')}} | Group-Object -Property 'UpnPresent' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize


# Group by
## UPN
$SignIns | Group-Object -Property 'UserPrincipalName' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## Alternate Sign In Name
$SignIns | Group-Object -Property 'AlternateSignInName' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## Risk level
$SignIns | Group-Object -Property 'RiskLevelDuringSignIn' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## IsInteractive
$SignIns | Group-Object -Property 'IsInteractive' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## App Id
$SignIns | Group-Object -Property 'AppId' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## ClientAppUsed
$SignIns | Group-Object -Property 'ClientAppUsed' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## AuthenticationDetails
### AuthenticationMethod
$SignIns | Select-Object -Property @{'Name'='AuthenticationMethod';'Expression'={$_.'AuthenticationDetails'.'AuthenticationMethod'}} | Group-Object -Property 'AuthenticationMethod' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### AuthenticationStepResultDetail
$SignIns | Select-Object -Property @{'Name'='AuthenticationStepResultDetail';'Expression'={$_.'AuthenticationDetails'.'AuthenticationStepResultDetail'}} | Group-Object -Property 'AuthenticationStepResultDetail' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### AuthenticationStepRequirement
$SignIns | Select-Object -Property @{'Name'='AuthenticationStepRequirement';'Expression'={$_.'AuthenticationDetails'.'AuthenticationStepRequirement'}} | Group-Object -Property 'AuthenticationStepRequirement' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### Succeeded
$SignIns | Select-Object -Property @{'Name'='Succeeded';'Expression'={$_.'AuthenticationDetails'.'Succeeded'}} | Group-Object -Property 'Succeeded' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## MfaDetail
### AuthMethod
$SignIns | Select-Object -Property @{'Name'='AuthMethod';'Expression'={$_.'MfaDetail'.'AuthMethod'}} | Group-Object -Property 'AuthMethod' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## DeviceDetail
### Browser
$SignIns | Select-Object -Property @{'Name'='Browser';'Expression'={$_.'DeviceDetail'.'Browser'}} | Group-Object -Property 'Browser' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### OperatingSystem
$SignIns | Select-Object -Property @{'Name'='OperatingSystem';'Expression'={$_.'DeviceDetail'.'OperatingSystem'}} | Group-Object -Property 'OperatingSystem' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## Status
### ErrorCode
$SignIns | Select-Object -Property @{'Name'='ErrorCode';'Expression'={$_.'Status'.'ErrorCode'}} | Group-Object -Property 'ErrorCode' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### FailureReason
$SignIns | Select-Object -Property @{'Name'='FailureReason';'Expression'={$_.'Status'.'FailureReason'}} | Group-Object -Property 'FailureReason' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
### AdditionalDetails
$SignIns | Select-Object -Property @{'Name'='AdditionalDetails';'Expression'={$_.'Status'.'AdditionalDetails'}} | Group-Object -Property 'AdditionalDetails' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize


# Analyze
## UPN on success authentications
$SignIns.Where{$_.'AuthenticationDetails'.'Succeeded'} | Group-Object -Property 'UserPrincipalName' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize
## UPN on failed authentications
$SignIns.Where{-not$_.'AuthenticationDetails'.'Succeeded'} | Group-Object -Property 'UserPrincipalName' -NoElement | Sort-Object -Property 'Count' -Descending | Format-Table -AutoSize


## Research
### Available properties
$SignIns | Select-Object -First 1 | Get-Member -MemberType '*Property' | Select-Object -ExpandProperty 'Name' | Sort-Object
$SignIns[0].RiskDetail
$SignIns[0].Status
$SignIns | Group-Object -Property Status -NoElement
$SignIns.Where{$_.RiskDetail}.Count
```

**Comment by christianarg** [comment]
@o-l-a-v 
1- how can I send you this information. I can't share it here as it list many UPN's from my organization
2- If I understand this correctly the commands you sent me are about user access to partner center (we're at 95% for users and we've already detected 2 signins that were made by mistake with an user with mfa exluded). Our main concern are "Through API or SDK", that we're at 39% altough the user that makes the consent has MFA enforced.

**Comment by o-l-a-v** [comment]
I'm not a Microsoft employee, so don't send me anything! :P

Im in the same boat as you, trying to find out what causes or bad score. I believe it's due to Cloudyn in our case. Also we use IT Glue, and their Office 365 integration is not using secure application model, so we disabled that as well.

I got this command from Yammer where I see you've also found. It certainly does not list the information we're looking for.

**Comment by christianarg** [comment]
@o-l-a-v yes I realised your'e not MS after I also found you on Yammer :)
Interesting, we also use Cloudyn, so that could be a cause. We'll give it a look. What are you going to do with cloudyn?

**Comment by isaiahwilliams** [comment]
@christianarg hopefully the percentages are looking better for you today. There have been a number of enhancements made to the command in question, so I would like to encourage you to update to the latest version and try again. The data returned from the Get-PartnerUserSignInActivity command will not match what is available in Partner Center due to the following reasons 

1. The values in the dashboard are percentages of requests not authentication attempt
2. Since the Azure AD audit log only contains records of authentication attempts you will not easily be able to determine if the secure application model has been done correctly. This is because the exchange of a refresh token for an access token is not captured in the Azure AD audit logs

Basically the data returned from this command will only obtain sign-in attempts. With respect to the API this will allow you to identify any that is connection using the auth code or password flow. Hopefully with this information you will be able to better leverage this command to perform any further investigation you need. If you are still having issues identifying the reason why the API/SDK percentages are low for your organization, then I would like to encourage you to work with your SAM/TAM to investigate further.  

If you are having issues with this module please let us know.
